### PR TITLE
Eng 8562 sqlcov patterns no queries

### PR DESCRIPTION
**Pull request by johnTcrawford**
These changes should eliminate all cases of patterns generating no SQL queries, for all SqlCoverage test suites, including in the "extended" tests. (Note: I will not merge until Jenkins passes, both the regular tests for this branch, and the extended SqlCoverage test run I kicked off.)

### CODE REVIEW

**Comment by BillWhite** [comment]
I'm not really sure what's going on here, but it seems plausible from what I understand.

File: tests/scripts/examples/sql_coverage/advanced-template.sql [hunk 1]
```diff
@@ -83,7 +83,7 @@ SELECT 20,                                               A._variable[#GB @column
 SELECT 21,                           @optionalfn(        A._variable[#GB @columntype] ) , @agg(            A._variable[@columntype] )        FROM @fromtables A GROUP BY         A.__[#GB]
 SELECT 22,                           @optionalfn(@onefun(A._variable[#GB @columntype])) , @agg(            A._variable[@columntype] )        FROM @fromtables A GROUP BY @onefun(A.__[#GB])
 -- multiple column group by
-SELECT 23,               A._variable[#GB1 @columntype],  A._variable[#GB2 @columntype]   , @agg(           A._variable[@columntype])         FROM @fromtables A GROUP BY         A.__[#GB1], A.__[#GB2]
+SELECT 23,            A._variable[#GB1 @comparabletype], A._variable[#GB2 @comparabletype], @agg(          A._variable[@comparabletype])     FROM @fromtables A GROUP BY         A.__[#GB1], A.__[#GB2]
```
Comment (paulmartel) [inline tests/scripts/examples/sql_coverage/advanced-template.sql]: Went too far with this one? I think that it's best practice to distinguish at least one _variable as having @columntype vs. @comparabletype. This prevents total redundancy of a pattern between configs if we run with identical schema but define a different numeric @columntype that has the same set of @comparabletypes as other numerics.

File: tests/scripts/examples/sql_coverage/advanced-template.sql [hunk 2]
```diff
@@ -91,7 +91,7 @@ SELECT     @agg(                     @optionalfn(        A._variable[@columntype
 -- group by alias (50 - 55)
 SELECT 50,                           @optionalfn(@onefun(A._variable[#GB @columntype])) as tag , @agg(            A._variable[@columntype] )        FROM @fromtables A GROUP BY tag
 SELECT 51,                           A._variable[#GB @columntype] as tag ,                       @agg(            A._variable[@columntype] )        FROM @fromtables A GROUP BY tag
-SELECT 52,              @optionalfn(@onefun(A._variable[#GB @columntype])) as tag1,  A._variable[@columntype] as tag2,  @agg(           A._variable[@columntype])         FROM @fromtables A GROUP BY        tag2, tag1
+SELECT 52, @optionalfn(@onefun(A._variable[#GB @comparabletype])) as tag1, A._variable[@comparabletype] as tag2, @agg(A._variable[@comparabletype]) FROM @fromtables A GROUP BY tag2, tag1
```
Comment (paulmartel) [inline tests/scripts/examples/sql_coverage/advanced-template.sql]: Again, no use of @columntype.

File: tests/scripts/examples/sql_coverage/configure-for-timestamp.sql [hunk 3]
```diff
@@ -5,11 +5,9 @@
 {@agg = "_genericagg"} -- timestamps don't do SUM or AVG
 {@columnpredicate = "_timestampcolumnpredicate"}
 {@columntype = "timestamp"}
--- OK. comparableconstant is not a constant, here.
--- just fudging a type-friendly placeholder until ad hoc timestamp composition is supported
-{@comparableconstant = "_variable[#reusable timestamp]"}
+{@comparableconstant = "'1967-10-01 15:25:26.123457'"}
```
Comment (paulmartel) [inline tests/scripts/examples/sql_coverage/configure-for-timestamp.sql]: yes. well done.

File: tests/scripts/examples/sql_coverage/configure-for-timestamp.sql [hunk 4]
```diff
@@ -5,11 +5,9 @@
...
 {@comparabletype = "timestamp"}
--- OK. comparablevalue is not a constant generator, either.
+-- OK. comparablevalue is not a constant generator, here.
```
Comment (paulmartel) [inline tests/scripts/examples/sql_coverage/configure-for-timestamp.sql]: Unless I am misreading my own comment, it appears to be working around a SQLGenerator.py shortcoming that has been fixed "for ages" -- the ability of _value[timestamp] to generate properly formatted date strings within a workable range. Maybe it's time to re-enable this canonical definition.

File: tests/scripts/sql_coverage_test.py [hunk 5]
```diff
@@ -440,7 +440,7 @@ def get_numerical_html_table_element(value, error_below=-1, strong_warn_below=0,
              value) )
 
 def get_time_html_table_element(seconds, error_below=0, strong_warn_below=0, warn_below=0,
-                                error_above=21600, strong_warn_above=3600, warn_above=600,  # 6 hours, 1 hour, 10 minutes
+                                error_above=28800, strong_warn_above=3600, warn_above=600,  # 8 hours, 1 hour, 10 minutes
```
Comment (paulmartel) [inline tests/scripts/sql_coverage_test.py]: Maybe? we can clamp back down on this when @columntype gets judiciously applied in statements that no longer reference it -- or maybe that will not significantly reduce redundant output -- maybe the day of the under 6 hour run is just gone.

**Comment by paulmartel** [comment]
end of round.

File: tests/scripts/examples/sql_coverage/advanced-template.sql (hunk 1 above)
Comment (johnTcrawford) [inline tests/scripts/examples/sql_coverage/advanced-template.sql]: OK, changed the first one from @comparabletype back to @columntype.

File: tests/scripts/examples/sql_coverage/advanced-template.sql (hunk 2 above)
Comment (johnTcrawford) [inline tests/scripts/examples/sql_coverage/advanced-template.sql]: Again, changed the first one from @comparabletype back to @columntype.

File: tests/scripts/examples/sql_coverage/configure-for-timestamp.sql (hunk 4 above)
Comment (johnTcrawford) [inline tests/scripts/examples/sql_coverage/configure-for-timestamp.sql]: Good idea. I uncommented this line:
{@comparablevalue = "_value[timestamp]"}
and deleted this one (as well as obsolete comments):
{@comparablevalue = "_variable[timestamp]"}
and it seems to work fine (for basic-timestamp, which I think is the only test suite that uses this).

File: tests/scripts/sql_coverage_test.py (hunk 5 above)
Comment (johnTcrawford) [inline tests/scripts/sql_coverage_test.py]: FYI, the only thing I changed here (from 6 to 8 hours) is the limit above which the time for a test suite (or part of it) will turn *red*, which normally means an error - a mismatch or crash; it is still orange, meaning a "strong warning", for anything over an hour, which seems sufficient - there are very few of those.

**Pull request merged**